install_files/ansible-base/roles/ossec-server/templates/send_encrypted_alarm.sh

```diff
@@ -1,5 +1,49 @@
 #!/bin/bash
+# Handler script to encrypt OSSEC alerts prior to mailing.
+# Called via the `.procmailrc` for user `ossec`.
 
-/usr/bin/formail -I "" | \
-    /usr/bin/gpg --homedir /var/ossec/.gnupg --trust-model always -ear "{{ ossec_gpg_fpr }}" | \
-    /usr/bin/mail -s "$(echo $SUBJECT | sed -r 's/([0-9]{1,3}\.){3}[0-9]{1,3}\s?//g' )" {{ ossec_alert_email }}
+# Intentionally NOT setting `set -e`, since we'll do our own error handling.
+# Rather than exit immediately on errors, we should gracefully inform
+# the Admin that there was a problem encrypting a message.
+#set -e
+set -o pipefail
+
+
+# Read alert message content from STDIN, as passed by OSSEC. Store the message
+# in a var so we can iteratively process it, halting execution before sending
+# if the GPG encryption command failed. If we simply chain a bunch of pipes
+# together, then the email will be sent with an empty body, should the GPG
+# encryption fail.
+ossec_alert_text="$(< /dev/stdin)"
+
+# Primary "send email to Admin" functionality.
+function send_encrypted_alert() {
+
+    local encrypted_alert_text
+    # Try to encrypt the alert message. We'll inspect the exit status of the
+    # pipeline to decide whether to send the alert text, or the default
+    # failure message.
+    encrypted_alert_text="$(printf "${ossec_alert_text}" | \
+        /usr/bin/formail -I '' | \
+        /usr/bin/gpg --homedir /var/ossec/.gnupg --trust-model always -ear '{{ ossec_gpg_fpr }}')"
+
+    # Error handling.
+    if [[ -z "${encrypted_alert_text}" || $? -ne 0 ]]; then
+        send_plaintext_fail_message
+    else
+        echo "${encrypted_alert_text}" | \
+            /usr/bin/mail -s "$(echo $SUBJECT | sed -r 's/([0-9]{1,3}\.){3}[0-9]{1,3}\s?//g' )" '{{ ossec_alert_email }}'
```

Comment:
The regular expression was added as part of #915, to scrub IP addresses from the subject line (which is observable in plain text, since it's outside the encrypted message body).